api/server/httprest/server.go

```diff
@@ -0,0 +1,105 @@
+package httprest
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v5/runtime"
+)
+
+var _ runtime.Service = (*Server)(nil)
+
+// httpHandler is a functional interface that implements the http handler functionality.
+type httpHandler func(
+	h http.HandlerFunc,
+	w http.ResponseWriter,
+	req *http.Request,
+)
+
+// Config parameters for setting up the http-rest service.
+type config struct {
+	httpAddr string
+	handler  httpHandler
+	router   *mux.Router
+	timeout  time.Duration
+}
+
+// Server serves HTTP traffic.
+type Server struct {
+	cfg          *config
+	server       *http.Server
+	cancel       context.CancelFunc
+	ctx          context.Context
+	startFailure error
+}
+
+// New returns a new instance of the Server.
+func New(ctx context.Context, opts ...Option) (*Server, error) {
+	g := &Server{
+		ctx: ctx,
+		cfg: &config{},
+	}
+	for _, opt := range opts {
+		if err := opt(g); err != nil {
+			return nil, err
+		}
+	}
+
+	if g.cfg.router == nil {
+		return nil, errors.New("router option not configured")
+	}
+
+	// TODO: actually use the timeout config provided
+	g.server = &http.Server{
+		Addr:              g.cfg.httpAddr,
+		Handler:           g.cfg.router,
+		ReadHeaderTimeout: time.Second,
+	}
+
+	return g, nil
+}
+
+// Start the http rest service.
+func (g *Server) Start() {
+	_, cancel := context.WithCancel(g.ctx)
+	g.cancel = cancel
```

Comment:
I don't think this is being used as intended. This cancel function will only cancel the new context which was thrown away.

Try this:

```suggestion
	g.ctx, g.cancel := context.WithCancel(g.ctx)
```